pkg/datasource/accessors.go

```diff
@@ -392,6 +395,10 @@ func (a *fieldAccessor) Tags() []string {
 	return slices.Clone(a.f.Tags)
 }
 
+func (a *fieldAccessor) AddTag(tag string) {
+	a.f.Tags = append(a.f.Tags, tag)
```

Comment:
Please lock this section like done with the other public functions:

```golang
	a.ds.lock.Lock()
	defer a.ds.lock.Unlock()
```

It seems like it's missing in some other functions as well - we need to address that as well, but shouldn't introduce more functions lacking it, IMHO.